store/store.go

```diff
@@ -141,9 +153,14 @@ func (s *Store[H]) Stop(ctx context.Context) error {
 	default:
 	}
 	// signal to prevent further writes to Store
-	s.writes <- nil
 	select {
-	case <-s.writesDn: // wait till it is done writing
+	case s.writes <- nil:
+	case <-ctx.Done():
+		return ctx.Err()
+	}
```

Comment:
This was a bug, and if we close the Store when writes are blocked it blocked the Close